jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/IOResources.java

```diff
@@ -90,6 +90,20 @@ public static RetainableByteBuffer toRetainableByteBuffer(Resource resource, Byt
             }
         }
 
+        // Optimize for Content.Source.Factory.
+        if (resource instanceof Content.Source.Factory factory)
```

Comment:
This check must come before calling `resource.getPath`, otherwise we cannot support `Resources` that return a `Path` for other reasons.

Reply:
Done.